cloudsql/sqlserver/database-sql/cloudsql.go

```diff
@@ -0,0 +1,306 @@
+		return votes, fmt.Errorf("DB.Query: %v", err)
+	}
+	defer rows.Close()
+	for rows.Next() {
+		nextVote := vote{}
+		err := rows.Scan(&nextVote.Candidate, &nextVote.VoteTime)
+		if err != nil {
+			return votes, fmt.Errorf("Rows.Scan: %v", err)
+		}
+		votes = append(votes, nextVote)
+	}
+	return votes, nil
+}
+
+// currentTotals returns a templateData structure for populating the web page.
+func currentTotals(app *App) (templateData, error) {
+	// get total votes for each candidate
+	var tabVotes, spaceVotes uint
+	err := app.db.QueryRow(`SELECT count(vote_id) FROM votes WHERE candidate='TABS'`).Scan(&tabVotes)
+	if err != nil {
+		return templateData{}, fmt.Errorf("DB.QueryRow: %v", err)
+	}
+	err = app.db.QueryRow(`SELECT count(vote_id) FROM votes WHERE candidate='SPACES'`).Scan(&spaceVotes)
+	if err != nil {
+		return templateData{}, fmt.Errorf("DB.QueryRow: %v", err)
+	}
+
+	var voteDiffStr string = voteDiff(int(math.Abs(float64(tabVotes) - float64(spaceVotes)))).String()
+
+	latestVotesCast, err := recentVotes(app)
+	if err != nil {
+		return templateData{}, fmt.Errorf("recentVotes: %v", err)
+	}
+	return templateData{tabVotes, spaceVotes, voteDiffStr, latestVotesCast}, nil
+
+}
+
+// showTotals renders an HTML template showing the current vote totals.
+func showTotals(w http.ResponseWriter, r *http.Request, app *App) error {
+	totals, err := currentTotals(app)
+	if err != nil {
+		return fmt.Errorf("currentTotals: %v", err)
+	}
+	err = parsedTemplate.Execute(w, totals)
+	if err != nil {
+		return fmt.Errorf("Template.Execute: %v", err)
+	}
+	return nil
+}
+
+// saveVote saves a vote passed as http.Request form data.
+func saveVote(w http.ResponseWriter, r *http.Request, app *App) error {
+	if err := r.ParseForm(); err != nil {
+		return fmt.Errorf("Request.ParseForm: %v", err)
+	}
+
+	var team string
+	if teamprop, ok := r.Form["team"]; ok {
+		team = teamprop[0]
+	} else {
+		return fmt.Errorf("team property missing from form submission")
+	}
+
+	// [START cloud_sql_sqlserver_databasesql_connection]
+	sqlInsert := "INSERT INTO votes (candidate, time_cast) VALUES (?, GETDATE())"
+	if team == "TABS" || team == "SPACES" {
+		if _, err := app.db.Exec(sqlInsert, team); err != nil {
+			fmt.Fprintf(w, "unable to save vote: %s", err)
+			return fmt.Errorf("DB.Exec: %v", err)
+		} else {
+			fmt.Fprintf(w, "Vote successfully cast for %s!\n", team)
+		}
+	}
+	return nil
+	// [END cloud_sql_sqlserver_databasesql_connection]
+}
+
+// mustGetEnv is a helper function for getting environment variables.
+// Displays a warning if the environment variable is not set.
+func mustGetenv(k string) string {
+	v := os.Getenv(k)
+	if v == "" {
+		log.Fatalf("Warning: %s environment variable not set.\n", k)
+	}
+	return v
+}
+
+// initSocketConnectionPool initializes a Unix socket connection pool for
+// a Cloud SQL instance of SQL Server.
+func initSocketConnectionPool() (*sql.DB, error) {
+	// [START cloud_sql_sqlserver_databasesql_create_socket]
```

Comment:
Any particular reason to not include the function declaration?

Reply:
The function declaration is omitted for consistency with the snippets from other languages. For example, see this Python snippet: https://cloud.google.com/sql/docs/mysql/manage-connections#pools-python